src/test/java/com/artipie/asto/S3StorageTest.java

```diff
@@ -71,6 +79,34 @@ void shouldUploadObjectWhenSave(final AmazonS3 client) throws Exception {
         MatcherAssert.assertThat(downloaded, Matchers.equalTo(data));
     }
 
+    @Test
+    void shouldFailToSaveMultipartUploadWhenFailedToReadContent(final AmazonS3 client) {
+        final String bucket = UUID.randomUUID().toString();
+        client.createBucket(bucket);
+        final String key = "fail";
+        final CompletableFuture<Void> saved = this.storage(bucket).save(
+            new Key.From(key),
+            new Content.From(Flowable.error(new IllegalStateException()))
+        );
+        Assertions.assertThrows(Exception.class, saved::join);
+    }
+
+    @Test
+    void shouldAbortMultipartUploadWhenFailedToReadContent(final AmazonS3 client) {
+        final String bucket = UUID.randomUUID().toString();
+        client.createBucket(bucket);
+        final String key = "abort";
+        final CompletableFuture<Void> saved = this.storage(bucket).save(
+            new Key.From(key),
+            new Content.From(Flowable.error(new IllegalStateException()))
+        );
```

Comment:
@olegmoz we can extract the save operation result (the `CompletableFuture`) in a private method that needs only the key as an arg. At least 3 tests are doing the same thing.

Reply:
@HDouss Do not think it is a good idea, because content we are trying to save is different. And `save` operation is what we are testing in here, so I'd like to keep it explicit.

Reply:
@olegmoz The point here is that bucket generation/creation is redundant. At least two tests have the same content. May be we can pass the Flowable as a second arg to this method.

Reply:
@HDouss In this methods - yes, `bucket` used only once. In other it is not the case. Bucket name used to create `S3Storage` and then used later to check what has happens after calling some `S3Storage` method.